### PR TITLE
code cleanup

### PR DESCRIPTION
**Pull request by dsyme**
some non-urgent code cleanup, will do some more soon

### CODE REVIEW

File: src/fsharp/CheckComputationExpressions.fs [hunk 1]
```diff
@@ -1180,20 +1179,40 @@ let TcComputationExpression cenv env (overallTy: OverallTy) tpenv (mWhole, inter
...
+
+            // Check if this is a Bind2Return etc.
+            let hasBindReturnN = not (isNil (TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AtMostOneResult cenv env bindRange ad bindReturnNName builderTy))
+            if hasBindReturnN && Option.isSome (convertSimpleReturnToExpr varSpace innerComp) then 
+                let consumePat = SynPat.Tuple(false, pats, letPat.Range)
+
+                // Add the variables to the query variable space, on demand
+                let varSpace = 
+                    addVarsToVarSpace varSpace (fun _mCustomOp env -> 
+                            use _holder = TemporarilySuspendReportingTypecheckResultsToSink cenv.tcSink
```
Comment (auduchinok) [inline src/fsharp/CheckComputationExpressions.fs:1205]: Should this line have less indentation?

File: src/fsharp/CheckComputationExpressions.fs [hunk 2]
```diff
@@ -1180,20 +1179,40 @@ let TcComputationExpression cenv env (overallTy: OverallTy) tpenv (mWhole, inter
...
-                let hasBindReturnN = not (isNil (TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AtMostOneResult cenv env bindRange ad bindReturnNName builderTy))
-                if hasBindReturnN && Option.isSome (convertSimpleReturnToExpr varSpace innerComp) then 
+            if not (cenv.g.langVersion.SupportsFeature LanguageFeature.AndBang) then
+                error(Error(FSComp.SR.tcAndBangNotSupported(), comp.Range))
+
+            if isQuery then
+                error(Error(FSComp.SR.tcBindMayNotBeUsedInQueries(), letBindRange))
+
+            let bindRange = match letSpBind with DebugPointAtBinding.Yes m -> m | _ -> letRhsExpr.Range
+            let sources = (letRhsExpr :: [for _, _, _, _, andExpr, _ in andBangBindings -> andExpr ]) |> List.map (mkSourceExprConditional isFromSource)
```
Comment (auduchinok) [inline src/fsharp/CheckComputationExpressions.fs:1189]: Maybe remove the parens?

File: src/fsharp/CheckComputationExpressions.fs [hunk 3]
```diff
@@ -1204,83 +1223,65 @@ let TcComputationExpression cenv env (overallTy: OverallTy) tpenv (mWhole, inter
...
-                        Some (transBind q varSpace bindRange bindNName sources consumePat letSpBind innerComp translatedCtxt)
-                    else
-
-                        // Look for the maximum supported MergeSources, MergeSources3, ... 
-                        let mkMergeSourcesName n = if n = 2 then "MergeSources" else "MergeSources"+(string n)
+                    let maxMergeSources =
+                        let rec loop (n: int) = 
+                            let mergeSourcesName = mkMergeSourcesName n
+                            if isNil (TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AtMostOneResult cenv env bindRange ad mergeSourcesName builderTy) then
+                                (n-1)
```
Comment (auduchinok) [inline src/fsharp/CheckComputationExpressions.fs:1235]: Let's add spaces here?

File: src/fsharp/CheckComputationExpressions.fs [hunk 4]
```diff
@@ -1180,20 +1179,40 @@ let TcComputationExpression cenv env (overallTy: OverallTy) tpenv (mWhole, inter
...
-
-                let numSources = sources.Length
-                let bindReturnNName = "Bind"+string numSources+"Return"
-                let bindNName = "Bind"+string numSources
-
-                // Check if this is a Bind2Return etc.
-                let hasBindReturnN = not (isNil (TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AtMostOneResult cenv env bindRange ad bindReturnNName builderTy))
-                if hasBindReturnN && Option.isSome (convertSimpleReturnToExpr varSpace innerComp) then 
+            if not (cenv.g.langVersion.SupportsFeature LanguageFeature.AndBang) then
+                error(Error(FSComp.SR.tcAndBangNotSupported(), comp.Range))
```
Comment (auduchinok) [inline src/fsharp/CheckComputationExpressions.fs:1183]: Shouldn't it have a space after `error`?

**Review by auduchinok — COMMENTED** [review]
Just a few questions about a preferred code style.

**Comment by dsyme** [comment]
> Just a few questions about a preferred code style.

THis was just un-indenting stuff, I wasn't cleaning up the style :))

I think we'll adjust all of the code of the compiler to match https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting

**Comment by auduchinok** [comment]
> THis was just un-indenting stuff, I wasn't cleaning up the style :))

Yeah, I hadn't realized it until wrote a bunch of comments. :)

**Comment by dsyme** [comment]
@TIHan @vzarytovskii @KevinRansom This is ready

**Review by TIHan — APPROVED** [review]

**Pull request merged**